File: scripts/generate-hero-webp.php
<?php
$heroDir = dirname(__DIR__) . '/frontend/window/img/hero';
$candidates = [
    'e978c0767c0fe7bc778596c86b2b54f3 1.png',
    'e978c0767c0fe7bc778596c86b2b54f3%201.png',
];
$heroPath = null;
foreach ($candidates as $f) {
    $p = $heroDir . DIRECTORY_SEPARATOR . $f;
    if (is_file($p)) {
        $heroPath = $p;
        break;
    }
}
if (!$heroPath) {
    fwrite(STDERR, "Hero PNG not found\n");
    exit(1);
}
if (!function_exists('imagewebp')) {
    fwrite(STDERR, "imagewebp not available\n");
    exit(1);
}
$img = @imagecreatefrompng($heroPath);
if (!$img) {
    fwrite(STDERR, "Failed to read PNG\n");
    exit(1);
}
$w = imagesx($img);
$h = imagesy($img);
echo "Source: {$w}x{$h}\n";
foreach ([640, 960, 1280, 1920] as $maxW) {
    $targetW = min($maxW, $w);
    $targetH = (int) round($h * ($targetW / $w));
    $resized = imagecreatetruecolor($targetW, $targetH);
    imagealphablending($resized, false);
    imagesavealpha($resized, true);
    imagecopyresampled($resized, $img, 0, 0, 0, 0, $targetW, $targetH, $w, $h);
    $out = $heroDir . '/home-hero-' . $maxW . '.webp';
    imagewebp($resized, $out, 78);
    imagedestroy($resized);
    echo basename($out) . ' ' . filesize($out) . " bytes\n";
}
imagedestroy($img);

File: scripts/generate_hero_webp.py
"""Generate responsive WebP hero images from source PNG."""
from pathlib import Path

from PIL import Image

ROOT = Path(__file__).resolve().parents[1]
HERO_DIR = ROOT / "frontend" / "window" / "img" / "hero"
CANDIDATES = [
    "e978c0767c0fe7bc778596c86b2b54f3 1.png",
    "e978c0767c0fe7bc778596c86b2b54f3%201.png",
]

source = next((HERO_DIR / name for name in CANDIDATES if (HERO_DIR / name).is_file()), None)
if source is None:
    raise SystemExit("Hero PNG not found")

img = Image.open(source).convert("RGB")
print(f"Source: {source.name} {img.size[0]}x{img.size[1]}")

for width in (640, 960, 1280, 1920):
    ratio = min(1.0, width / img.size[0])
    target = (int(img.size[0] * ratio), int(img.size[1] * ratio))
    resized = img if ratio >= 1 else img.resize(target, Image.Resampling.LANCZOS)
    out = HERO_DIR / f"home-hero-{width}.webp"
    resized.save(out, "WEBP", quality=78, method=6)
    print(f"{out.name}: {out.stat().st_size // 1024} KB")
